### PR TITLE
Added new function for randomized recommended recipes and additional …

### PR DESCRIPTION
**Pull request by Sedwards8900**
"Added function for randomized recommended recipes within spoonacular.py file and added GoogleUser model and route changes to login, search, and index for testing spoonacular newly created function"

### CODE REVIEW

**Review by Sedwards8900 — COMMENTED** [review]
I am self-submitting changes to address merge conflicts with recent pulls submitted yesterday and my latest pull version into my personal repository, in addition to adding new method to spoonacular

File: src/database/models.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 from sqlalchemy.orm import relationship
 from database.database import DatabaseException
 from datetime import datetime
+from flask_login import UserMixin
```
Comment (williamgarland) [inline src/database/models.py:10]: This was already imported below
```suggestion
```

File: src/routes/index.py [hunk 2]
```diff
@@ -1,6 +1,8 @@
 from flask import Blueprint, render_template
 from flask_login import current_user
 from . import util
+from flask_login import current_user
```
Comment (williamgarland) [inline src/routes/index.py:4]: This was already imported on line 2
```suggestion
```

File: src/routes/login.py [hunk 3]
```diff
@@ -56,7 +53,7 @@ def login():
...
 
 
-@login_blueprint.route("/logout")
-def logout():
-    logout_user()
-    return redirect("/")
+# @login_blueprint.route("/logout")
+# def logout():
+#     logout_user()
+#     return redirect("/")
```
Comment (williamgarland) [inline src/routes/login.py:59]: Why was this commented out?

File: src/routes/index.py [hunk 4]
```diff
@@ -24,7 +26,17 @@ def get_blueprint():
...
+            "TESTING AUTH SECTION" + current_user.name,
+            current_user.email,
+            current_user.profile_pic,
+        )
+        # return render_template("index2.html", userdata=current_user.to_json())
+    # Else, get dummy data/random recommendations
     else:
-        return render_template("index.html")
+        data = get_recommended_recipes()
+        print(data)
```
Comment (williamgarland) [inline src/routes/index.py:40]:
```suggestion
```

File: src/api/spoonacular.py [hunk 5]
```diff
@@ -474,6 +474,55 @@ def search_ingredients(
...
+
+    params = {"apiKey": get_api_key()}
+
+    params["offset"] = offset
+    params["number"] = limit
+
+    data = None
+    try:
+        data = api_get_json(
+            SPOONACULAR_API_ROOT_ENDPOINT + f"recipes/random?",
```
Comment (williamgarland) [inline src/api/spoonacular.py:507]: I dont think the '?' is necessary in the url
```suggestion
            SPOONACULAR_API_ROOT_ENDPOINT + "recipes/random",
```

File: src/api/spoonacular.py [hunk 6]
```diff
@@ -474,6 +474,55 @@ def search_ingredients(
...
 
+def get_recommended_recipes(
+    offset: int = 0,
+    limit: int = 10,
+) -> list:
+    """
+    Returns a list of randomly recommended recipes.
+
+    Args:
+        Number as the number of desired recipe results
```
Comment (williamgarland) [inline src/api/spoonacular.py:485]: This is not descriptive of what arguments this function accepts
```suggestion
        offset (int): The offset into the search results to start at.
        limit (int): The maximum number of search results to retrieve.
```

**Review by williamgarland — CHANGES_REQUESTED** [review]

File: src/routes/index.py (hunk 4 above)
Comment (Sedwards8900) [reply src/routes/index.py:40]: This print statement is for testing how data will be displayed

**Comment by Sedwards8900** [comment]
Requested changes have been performed successfully.

**Pull request merged**